fabriziogianni7/botblock
# feat(protect): add basic protect landing page

## Description

This PR initiates the user's (Content Creator) path to secure their website from bots. To do so, they browse to the `protect/landing` and enter their site's URL.

From there, the app retrieves the `robots.txt` file (as developed on #4) and then redirects the user to the next step on a different screen.

Since information should be persisted across screens, a `RobotsContext` was developed to store that info and have it accessible by multiple components and pages.

Lastly, some rough styling was applied to the landing screen but further tweaks shall be made down the road.

# How to test

1. Start express server by running `yarn start` on `./packages/express`
3. Start nextjs by running `yarn dev` on `./packages/nextjs`
4. Browse to `protect/landing`
5. Enter some valid url to fetch its `robots.txt` from (i.e. `https://www.google.com`)
6. Click submit
7. Check that after loading, the user is redirected to `protect/robots-txt` dummy page.

## Related links

- Closes #2 

## Proof

| _Demo_ |
| :---: |
| <video src='https://github.com/fabriziogianni7/botblock/assets/21087992/cdbb5f31-06cd-4af5-b1c2-56a3b4da6db6' > |
